docs/oauth2.md

```diff
@@ -62,149 +71,79 @@ auth_backends.1 = rabbit_auth_backend_oauth2
-```
+The next sections cover in more detail what happens during authentication and how to configure OAuth 2.0 beyond the basic configuration shown above.
 
-it will translate into the following configuration (in the [advanced RabbitMQ config format](./configure)):
+### Authorization Flow {#authorization-flow}
 
-```erlang
-[
-  %% ...
-  %% backend configuration
-  {rabbitmq_auth_backend_oauth2, [
-    {resource_server_id, <<"my_rabbit_server">>},
-    %% UAA signing key configuration
-    {key_config, [
-      {signing_keys, #{
-        <<"a-key-ID">> => {pem, <<"-----BEGIN PUBLIC KEY-----
-MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA2dP+vRn+Kj+S/oGd49kq
-6+CKNAduCC1raLfTH7B3qjmZYm45yDl+XmgK9CNmHXkho9qvmhdksdzDVsdeDlhK
-IdcIWadhqDzdtn1hj/22iUwrhH0bd475hlKcsiZ+oy/sdgGgAzvmmTQmdMqEXqV2
-B9q9KFBmo4Ahh/6+d4wM1rH9kxl0RvMAKLe+daoIHIjok8hCO4cKQQEw/ErBe4SF
-2cr3wQwCfF1qVu4eAVNVfxfy/uEvG3Q7x005P3TcK+QcYgJxav3lictSi5dyWLgG
-QAvkknWitpRK8KVLypEj5WKej6CF8nq30utn15FQg0JkHoqzwiCqqeen8GIPteI7
-VwIDAQAB
------END PUBLIC KEY-----">>}
-          }}
-      ]}
-    ]}
-].
-```
+This plugin does not communicate with any OAuth 2.0 provider in order to authenticate user and grants access. Instead, it decodes an access token provided by the client and authorises a user based on the scopes found in the token.
 
-If a symmetric key is used, the configuration will look like this:
+Tokens must be digitally signed in order to be accepted. RabbitMQ must have the signing key to validate the signature. You can either configure the signing keys the OAuth 2.0 provider will use or configure RabbitMQ with one of the two endpoints below:
 
-```erlang
-[
-  {rabbitmq_auth_backend_oauth2, [
-    {resource_server_id, <<"my_rabbit_server">>},
-    {key_config, [
-      {signing_keys, #{
-        <<"a-key-ID">> => {map, #{<<"kty">> => <<"MAC">>,
-                                  <<"alg">> => <<"HS256">>,
-                                  <<"value">> => <<"my_signing_key">>}}
-      }}
-    ]}
-  ]},
-].
-```
+  - **JWKS endpoint** - this is the HTTP endpoint that returns the signing keys used to digitally sign the tokens.
+  - **OpenID Provider Configuration endpoint** - this is the endpoint that returns the provider's configuration including all its endpoints, such as the **JWKS endpoint**.
 
-The key set can also be retrieved dynamically from a URL serving a [JWK Set](https://tools.ietf.org/html/rfc7517#section-5).
-In that case, the configuration will look like this:
+When you configure RabbitMQ with one these URLs, RabbitMQ must make a HTTP request (or two, if we specify the latter endpoint) to download the signing keys. This is an operation that occurs once for any signing key not downloaded yet. When the OAuth 2.0 provider rotates the signing keys, newer tokens refer to a new signing key which RabbitMQ does not have yet which triggers another download of the newer signing keys.
```

Comment:
"one of these URLs"?? what URLs are you referring to here? 

"which triggers another download......" is that trigger automatic because new tokens refer to new signing keys?

Reply:
Correct , when the user tries to authenticate with a token which was signed with a signing key that RabbitMQ has not ever downloaded, RabbitMQ has to download it .. that is why i say .. "triggers another download .. "